src/types.ts

```diff
@@ -464,3 +467,58 @@ export type IEmailGetResponse = IGetResponse<IEmailProperties>;
 export type IEmailSetArguments = ISetArguments<IEmailProperties>;
 
 export type IEmailSetResponse = ISetResponse<IEmailProperties>;
+
+/**
+ * See https://jmap.io/spec-mail.html#email-submission
+ */
+export interface DeliveryStatus {
+  smtpReply: string;
+  delivered: 'queued' | 'yes' | 'no' | 'unknown';
+  displayed: 'unknown' | 'yes';
+}
+
+/**
+ * See https://jmap.io/spec-mail.html#email-submission
+ */
+export interface Address {
+  email: string;
+  parameters: any | null;
+}
+/**
+ * See https://jmap.io/spec-mail.html#email-submission
+ */
+export interface Envelope {
+  mailFrom: Address;
+  rcptTo: Address[];
+}
+/**
+ * See https://jmap.io/spec-mail.html#email-submission
+ */
+export interface IEmailSubmissionProperties {
+  id: string;
+  identityId: string;
+  emailId: string;
+  threadId: string;
+  envelope: Envelope | null;
+  sendAt: Date;
```

Comment:
I wonder if we would deserve a type to distinguish `Date` from `UTCDate`... Here it should be UTCDate.

Cd https://jmap.io/spec-core.html#the-date-and-utcdate-data-types

Reply:
There is one here https://github.com/OpenPaaS-Suite/jmap-client-ts/blob/main/src/types.ts#L246. So the only option is to create separate one for string, to avoid confusion.